examples/src/main/java/com/squareup/moshi/recipes/ReadJsonString.java

```diff
@@ -0,0 +1,64 @@
+package com.squareup.moshi.recipes;
+
+import com.squareup.moshi.FromJson;
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonQualifier;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import com.squareup.moshi.Moshi;
+import com.squareup.moshi.ToJson;
+import java.io.IOException;
+import java.lang.annotation.Retention;
+import okio.Buffer;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+final class ReadJsonString {
+  @Retention(RUNTIME)
+  @JsonQualifier
+  public @interface JsonString {
+    Object ADAPTER = new Object() {
+      @FromJson @JsonString String fromJson(JsonReader reader) throws IOException {
+        return reader.readJsonString().utf8();
+      }
+
+      @ToJson void toJson(JsonWriter writer, @JsonString String value) throws IOException {
+        writer.value(new Buffer().writeUtf8(value));
+      }
+    };
+  }
+
+  static final class Location {
+    final String name;
+    final @JsonString String metadata;
```

Comment:
honest question: should we encourage users to store these as ByteStrings? If it’s an opaque blob it might as well be?